editor/index.js

```diff
@@ -107,7 +108,9 @@ export function createEditorInstance( id, post, editorSettings = DEFAULT_SETTING
 						onUndo: undo,
 					}, store.dispatch ) }
 				>
-					<Layout settings={ editorSettings } />
+					<EditorSettingsProvider settings={ editorSettings }>
```

Comment:
yeah, maybe `composeProviders` or something :)

Maybe if we extract the editor, we could reduce the number of providers into one (inside the generic editor component) and provide an HoC like `withEditorData` instead of `withEditorSettings` to retrieve the editor settings/editor local state from any component. (just an idea we could explore later, not sure if it's a good one yet)